tests/integration_tests/security/spectre-mitigations

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
This test is useful for users verifying their integration meets our recommendations and for verifying our own testing/benchmarking environments meet these recommendations.

About this I would first ask, do we not use the same machines for CI as for performance tests? and secondly, do we not already run integration tests before beginning these performance tests?

I believe for these longer running performance tests it would be advisable to run all integration tests at the start. Taking this approach I believe is preferable to manual selection of specific tests and guarantees we are benchmarking a fully working executable as we intend it. If we do not already run integration tests before long running performance tests I would suggest this as another PR.

Reply:
I do not think it is necessary to run these test on our CI AMIs as we already know these settings are enabled by default (I believe we also do not validate other default system settings we recommend). But I'm not very familiar here so would also appreciate additional input.

Reply:
> I disagree with the notion that users should run our integration test in a production machine (where the security recommendations matter). Very often users just run in production pre-built binaries from a release, no need to build Firecracker there.
> 
> Having a tool to check production machines compliance to our recommendations is useful but I don't think our integration tests is the right tool to do so.

Relating to these points I would suggest that as we offer a pre-built binaries Firecracker we could also offer pre-built binaries for the Rust tests. This way a user could simply run this in the same environment as the Firecracker application binary. Among additional benefits believe this also efficiently fulfills the task of a tool to check machine compliance to our recommendations. Notably this could not  (or I am not aware of a simple way to) include Python tests, another motivation perhaps to move towards Rust for these integration tests.

Reply:
So the goal as I interpret it is that whenever we make performance tests, we should be running SPECTRE mitigations. Ideally we would check everything is enabled when we create the testing images, but right now we don't have a way to do this, so the next best thing is to run them every time we run integration tests.

I agree that this test is a bit of a kludge, as we know the images are already well configured, so in a sense this test is working as a canary, making sure we will notice if something changes. In the future when we can run validation during image creation we should move this test there and remove it from here.